src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs

```diff
@@ -114,30 +113,33 @@ private async Task<HttpResponseMessage> SendInvocationRequestAsync(ScriptInvocat
             if (_enableRequestTracing)
             {
                 scriptInvocationContext.Logger.LogTrace($"Invocation Request:{httpRequestMessage}");
-                await TraceHttpContent(httpRequestMessage.Content, scriptInvocationContext.Logger);
+                await GetHttpContentAsString(httpRequestMessage.Content, scriptInvocationContext.Logger);
             }
             _logger.CustomHandlerSendingInvocation(scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
             HttpResponseMessage invocationResponse = await _httpClient.SendAsync(httpRequestMessage);
             if (_enableRequestTracing)
             {
                 scriptInvocationContext.Logger.LogTrace($"Invocation Response:{invocationResponse}");
-                await TraceHttpContent(invocationResponse.Content, scriptInvocationContext.Logger);
+                await GetHttpContentAsString(invocationResponse.Content, scriptInvocationContext.Logger);
             }
             _logger.CustomHandlerReceivedInvocationResponse(scriptInvocationContext.FunctionMetadata.Name, scriptInvocationContext.ExecutionContext.InvocationId);
             return invocationResponse;
         }
 
-        private static async Task TraceHttpContent(HttpContent content, ILogger logger)
+        private static async Task<string> GetHttpContentAsString(HttpContent content, ILogger logger = null)
         {
             if (content != null)
             {
                 bool isMediaTypeOctetOrMultipart = content.Headers != null && Utility.IsMediaTypeOctetOrMultipart(content.Headers.ContentType);
                 // do not log binary data as string
                 if (!isMediaTypeOctetOrMultipart)
                 {
-                    logger.LogTrace(await content.ReadAsStringAsync());
+                    string stringCotent = await content.ReadAsStringAsync();
+                    logger?.LogTrace(stringCotent);
```

Comment:
Where is this log going? It won't go to system logs, right?

Reply:
This is user logger. When running locally, request/response logged to console.